src/presentational-components/shared/table-composable-toolbar-view.tsx

```diff
@@ -0,0 +1,367 @@
+import React, { Fragment } from 'react';
+import { useIntl } from 'react-intl';
+import messages from '../../Messages';
+import { TableVariant, TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import TableToolbar from '@redhat-cloud-services/frontend-components/TableToolbar';
+import { Button, Pagination, EmptyStatePrimary } from '@patternfly/react-core';
+import { ListLoader } from './loader-placeholders';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import Toolbar, { paginationBuilder } from './toolbar';
+import EmptyWithAction from './empty-state';
+import './table-toolbar-view.scss';
+import { ISortBy, OnSort } from '@patternfly/react-table';
+import { CellObject, RowProps } from '../../smart-components/user/user-table-helpers';
+
+interface FilterProps {
+  username?: string;
+  email?: string;
+  status?: string;
+}
+
+interface PaginationProps {
+  limit?: number;
+  offset?: number;
+  count?: number;
+  noBottom?: boolean;
+}
+
+// TODO: make it clear later and remove FetchDataProps from below
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface FetchDataPropsNew {
+  pagination?: PaginationProps;
+  filters?: FilterProps;
+  orderBy?: string;
+}
+
+interface FetchDataProps {
```

Comment:
ALso here we could extend PaginationProps

Reply:
I removed PaginationProps for now, it was not used